apps/recnet-api/src/utils/pipes/zod.validation.pipe.ts

```diff
@@ -4,13 +4,23 @@ import { ZodError, ZodSchema } from "zod";
 import { RecnetError } from "@recnet-api/utils/error/recnet.error";
 import { ErrorCode } from "@recnet-api/utils/error/recnet.error.const";
 
+export type MetadataType = "custom" | "body" | "query" | "param" | "header";
+
 export class ZodValidationPipe implements PipeTransform {
-  constructor(private schema: ZodSchema) {}
+  constructor(
+    private schema: ZodSchema,
+    private metadataType?: MetadataType
+  ) {}
 
   transform(value: unknown, metadata: ArgumentMetadata) {
     if (metadata.type === "custom") {
       return value;
     }
+
+    if (this.metadataType && metadata.type !== this.metadataType) {
+      return value;
+    }
+
```

Comment:
Why would we need this?

Reply:
In `PATCH /announcement/:id` API, the body validator will be applied to parameter. As a result, I added this as a workaround to resolve this issue.

Reply:
We will create another PR to make validationPipe API easier to use and understand.